woocommerce/woocommerce
# Deriving product description from product title and removing advanced form

### Submission Review Guidelines:

- I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
- I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
- I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/). 
- Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:

Refactoring client-side code, simplifying to reduce complexity in removing advanced form (for now) and deriving the description only from the product title. There is also some experimentation with the building the prompt as a starting point.

Closes #38337 .



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Checkout branch and link `woo-ai` plugin to your local.
2. Activate the `woo-ai` plugin.
3. Ensure your local has the `OPEN_AI_KEY` defined.
4. Navigate to Products -> Add New
5. Notice that the "Write it for me" button is disabled before you enter a title.
6. Hover your mouse over the disabled button to see a bit of helper text.
7. Enter a product title.
8. Click the now-enabled "Write it for me" button to generate a description.
